### PR TITLE
feat: sign-in page, icons, and styling

### LINKED ISSUE #189: Hubspot signup

### PR DESCRIPTION
**Pull request by thantos**
Closes #189 
Part of #204

* Hubspot signup form on /sign-up
* `FunctionlessHighlighter`
* Start using material UI (MUI)
* theming and dark mode
* Updated Icon(s)

### CODE REVIEW

File: website/src/components/highlighter.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+ * Function - Generally classified as a built in, this highlighter removes the built-in
+ * to support Functionless {@link Function}.
+ */
+export function FunctionlessHighlighter({
+  children,
+  tokenReplacement,
+}: HighlighterProps) {
+  return (
+    <Highlighter
+      tokenReplacement={(token) => {
```
Comment (sam-goodwin) [inline website/src/components/highlighter.tsx:28]: Nice one. I am impressed you actually solved it haha.

**Review by sam-goodwin — COMMENTED** [review]
Super impressed.

**Comment by sam-goodwin** [comment]
Would love to see the netlify preview. Build is broken?

**Comment by sam-goodwin** [comment]
V8 crashed it seems. Weird

File: website/src/components/highlighter.tsx (hunk 1 above)
Comment (thantos) [reply website/src/components/highlighter.tsx:28]: Everything is an AST. All I see anymore are ASTs.

**Comment by thantos** [comment]
Works locally, looks like we ran out of memory, any idea how much memory the netlify host has?

**Comment by thantos** [comment]
Apparently I just had to increase the node memory, maybe netlify has lower defaults? I think its settable in the netlify account too, but setting from env variable works for now.

**Comment by sam-goodwin** [comment]
I'm surprised netlify is taking 10min to build. Are we building and testing functionless when building the website?

**Comment by thantos** [comment]
Nope, just `website:build`, but docusuarus and webpack are slow

**Comment by thantos** [comment]
with more memory it built in 6 minutes! (set the max memory to 8gb in the env variables on netlify)

**Comment by sam-goodwin** [comment]
We should add the following to the list of IaC frameworks:
- [ ] Terraform CDK
- [ ] Azure Resource Manager
- [ ] Serverless Stack

File: website/package.json [hunk 2]
```diff
@@ -21,18 +21,25 @@
...
+    "@emotion/react": "^11.9.0",
+    "@emotion/styled": "^11.8.1",
+    "@mui/icons-material": "^5.8.2",
+    "@mui/material": "^5.8.2",
     "@tsconfig/docusaurus": "^1.0.5",
     "@types/react": "^17.0.0",
     "@types/react-syntax-highlighter": "^15.5.1",
     "docusaurus-plugin-typedoc": "^0.17.4",
+    "install": "^0.13.0",
+    "npm": "^8.12.1",
```
Comment (sam-goodwin) [inline website/package.json:41]: hmm, that's interesting. Why do we need this explicitly?

**Review by sam-goodwin — APPROVED** [review]

**Comment by sam-goodwin** [comment]
Love the green text on dark
image


Maybe a bit hard to read on light mode?

image

**Comment by sam-goodwin** [comment]
Can we replace the hardcore green on the front page with a white or light grey? I think it's too in your face.

File: website/package.json (hunk 2 above)
Comment (thantos) [reply website/package.json:41]: Likely a mistake... Will fix.

**Comment by thantos** [comment]
image

**Comment by sam-goodwin** [comment]
Need to fix this. Is it the favicon?

image

**Comment by sam-goodwin** [comment]
> Need to fix this. Is it the favicon?
> 
> image

Fixed it. Look at mah UI skillz

**Comment by thantos** [comment]
> > Need to fix this. Is it the favicon?
> > 
> > image
> 
> Fixed it. Look at mah UI skillz

*Slow clap*

**Comment by thantos** [comment]
image

**Comment by thantos** [comment]
Linkedin
https://www.linkedin.com/post-inspector/inspect/https:%2F%2Fdeploy-preview-196--effortless-malabi-1c3e77.netlify.app%2Fdocs%2Fwhat-is-functionless%2F
Twitter
image

**Review by sam-goodwin — APPROVED** [review]

**Review by sam-goodwin — APPROVED** [review]

**Pull request merged**